model/pod/pod.go

```diff
@@ -0,0 +1,74 @@
+package pod
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Pod represents a related collection of containers. This model is hold
+// metadata about containers running in a container orchestration system.
+type Pod struct {
+	// ID is the unique identifier for the metadata document. This is
+	// semantically unrelated to the PodID.
+	ID string `bson:"_id" json:"id"`
+	// PodID is the unique resource identifier for the collection of containers
+	// running in the container orchestration service.
+	PodID string `bson:"pod_id" json:"pod_id"`
+	// TaskCreationOpts are options to configure how a task should be
+	// containerized and run in a pod.
+	TaskContainerCreationOpts TaskContainerCreationOptions `bson:"task_creation_opts,omitempty" json:"task_creation_opts,omitempty"`
+	// TimeInfo contains timing information for the pod's lifecycle.
+	TimeInfo TimeInfo `bson:"time_info" json:"time_info"`
```

Comment:
Should this be omitempty?

Reply:
Yeah, we can probably omitempty all the fields actually.

I also implemented `IsZero` for the `TimeInfo` and `TaskContainerCreationOptions` so that the BSON encoding knows when it's the zero value and can be omitted (from [the docs](https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#hdr-Structs)).